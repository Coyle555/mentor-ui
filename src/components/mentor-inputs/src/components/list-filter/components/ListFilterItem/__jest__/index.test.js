import React from 'react';
import { ListFilterItem } from '../index';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

afterEach(cleanup);

test('Default render of a list filter item', () => {
	const tree = renderer.create(<ListFilterItem />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Render of a list filter item with a custom class', () => {
	const tree = renderer.create(<ListFilterItem listClasses={{ item: 'foo' }} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Rendering a list filter item with a custom component', () => {
	const CustomComponent = (props) => <div>{JSON.stringify(props)}</div>;

	const tree = renderer.create(
		<ListFilterItem CustomListItem={CustomComponent} option="foo" />
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Moving mouse cursor over a list filter item', () => {
	const onMouseOver = jest.fn();
	const { getByText } = render(
		<ListFilterItem index={0} onMouseOver={onMouseOver} option="foo" />
	);

	fireEvent.mouseOver(getByText('foo'));
	expect(onMouseOver).toHaveBeenCalledWith(0);
});

test('Clicking on a list filter item', () => {
	const onClick = jest.fn();
	const { getByText } = render(
		<ListFilterItem onClick={onClick} option="foo" />
	);

	fireEvent.click(getByText('foo'));
	expect(onClick).toHaveBeenCalledWith('foo');
});
