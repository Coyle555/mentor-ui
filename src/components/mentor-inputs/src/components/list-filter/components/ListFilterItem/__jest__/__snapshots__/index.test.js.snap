// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render of a list filter item 1`] = `
<li
  className="apm-list-filter-menu-li"
  onClick={[Function]}
  onMouseOver={[Function]}
/>
`;

exports[`Render of a list filter item with a custom class 1`] = `
<li
  className="apm-list-filter-menu-li foo"
  onClick={[Function]}
  onMouseOver={[Function]}
/>
`;

exports[`Rendering a list filter item with a custom component 1`] = `
<li
  className="apm-list-filter-menu-li"
  onClick={[Function]}
  onMouseOver={[Function]}
>
  <div>
    {"option":"foo"}
  </div>
</li>
`;
